inspirehep/testlib/api/mitm_client.py

```diff
@@ -24,18 +24,72 @@
 
 from __future__ import absolute_import, division, print_function
 
+from decorator import decorate
+from os import environ
+
 from . import Session
 
 
+def with_mitmproxy(*args, **kwargs):
+    '''Decorator to abstract fixture recording and scenario setup for the E2E tests with mitmproxy.
+
+    Args:
+        scenario_name (Optional[str]): scenario name, by default test name without 'test_' prefix
+        should_record (Optional[bool]): is recording new interactions allowed during test run?
+        *args (List[Callable]): list of length of either zero or one: decorated function
+
+    Returns:
+        Callable: a decorator the can be used both with and without calling brackets
+            (if all params should be default)
+    '''
+    scenario_name = kwargs.pop('scenario_name', None)
+    should_record = kwargs.pop('should_record', False)
+
+    if not args:
+        assert not kwargs, 'Parameters %s not expected' % kwargs
+
+
+    def _with_mitmproxy(func, *args, **kwargs):
+        def ommit_test_in_name(name):
+            if name.startswith('test_'):
+                return name[5:]
+
+        mitmproxy_url = environ.get('MITMPROXY_HOST', 'http://mitm-manager.local')
+        mitm_client = MITMClient(mitmproxy_url)
+        final_scenario_name = scenario_name or ommit_test_in_name(func.__name__)
+
+        try:
+            mitm_client.set_scenario(final_scenario_name)
+            if should_record:
+                mitm_client.start_recording()
+            func(*args, **kwargs)
+        except:
+            raise
+        finally:
+            mitm_client.stop_recording()
+            mitm_client.set_scenario('default')
+
+    def _decorator(func):
+        return decorate(func, _with_mitmproxy)
+
+    if args:
+        return decorate(args[0], _with_mitmproxy)
+
+    return _decorator
+
+
 class MITMClient(object):
     def __init__(self, proxy_host='http://mitm-manager.local'):
         self._client = Session(base_url=proxy_host)
 
     def set_scenario(self, scenario_name):
-        self._client.put('/config', json={'active_scenario': scenario_name})
+        resp = self._client.put('/config', json={'active_scenario': scenario_name})
+        resp.raise_for_status()
```

Comment:
check if we want to return the resp.json here too

Reply:
`PUT /config` or `POST /config` don't return anything yet, and in fact PUT-ting triggers a `204 No Content` reply, so we might need to change that first (aand make the whole thing actual REST). In fact the response codes for `/record` endpoint are similar (despite returning data) so that an actual mistake in mitm-proxy which we should fix first.